src/main/java/uk/ac/ebi/ena/readtools/fastq/enafastq2sam/IlluminaBamConsumer.java

```diff
@@ -0,0 +1,190 @@
+/*
+ * Copyright 2010-2020 EMBL - European Bioinformatics Institute
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this
+ * file except in compliance with the License. You may obtain a copy of the License at
+ * http://www.apache.org/licenses/LICENSE-2.0
+ * Unless required by applicable law or agreed to in writing, software distributed under the
+ * License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR
+ * CONDITIONS OF ANY KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations under the License.
+ */
+package uk.ac.ebi.ena.readtools.fastq.enafastq2sam;
+
+import htsjdk.samtools.ReservedTagConstants;
+import htsjdk.samtools.SAMFileHeader;
+import htsjdk.samtools.SAMFileWriter;
+import htsjdk.samtools.SAMFileWriterFactory;
+import htsjdk.samtools.SAMReadGroupRecord;
+import htsjdk.samtools.SAMRecord;
+import uk.ac.ebi.ena.readtools.loader.common.InvalidBaseCharacterException;
+import uk.ac.ebi.ena.readtools.loader.common.QualityNormalizer;
+import uk.ac.ebi.ena.readtools.loader.common.eater.DataConsumer;
+import uk.ac.ebi.ena.readtools.loader.common.eater.DataConsumerException;
+import uk.ac.ebi.ena.readtools.loader.fastq.IlluminaSpot;
+
+import java.nio.file.Paths;
+import java.util.Set;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+import java.util.stream.Collectors;
+
+/**
+ * Accepts Illumina spot data and writes them out to a BAM file.
+ */
+public class IlluminaBamConsumer implements DataConsumer<IlluminaSpot, Object> {

```

Comment:
i think all these eaters/consumers have the word Illumina in them because they accept an object of IlluminaSpot class. I dont really know why is that class called so. Maybe because of the terminologies or format used inside it?

https://github.com/enasequence/readtools/blob/master/src/main/java/uk/ac/ebi/ena/readtools/loader/fastq/IlluminaSpot.java

Reply:
The IlluminaSpot is used to bring together paired reads. Better name would be 'FastqRead'. Spot is an alternative name for read on a physical experiment plate where the read is computed from a visual spot that changes intensity over time.